Make unlit path sample textures with nearest filtering

Noxime opened the pull request.

Noxime committed:
Make unlit path sample textures with nearest filtering

This is useful for drawing pixel art sprites

Noxime committed:
Merge branch 'trunk' of https://github.com/BVE-Reborn/rend3 into closest-filter

cwfitzgerald commented on rend3/shaders/lighting/pixel.glsl:
```diff
@@ -47,7 +47,11 @@ PixelData get_per_pixel_data(MATERIAL_TYPE material) {
     if (MATERIAL_FLAG(FLAGS_ALBEDO_ACTIVE)) {
         if (HAS_ALBEDO_TEXTURE) {
-            pixel.albedo = texture(sampler2D(ALBEDO_TEXTURE, linear_sampler), coords);
+            if (MATERIAL_FLAG(FLAGS_UNLIT)) {
```
On second thought we should just have a new material flag that determines which of the two samplers to use.

cwfitzgerald reviewed (changes requested):
Code looks fine, just a change of heart in the shader

cwfitzgerald commented:
Bump on this?

Noxime commented:
Oh sorry, I haven't worked at all on this, I've been busy with other stuff, and probably won't get around to this for next few days either. You are free to implement this yourself if you have time :)

cwfitzgerald commented:
No rush at all, just wanted to make sure you didn't forget :)

Noxime committed:
Merge branch 'trunk' of https://github.com/BVE-Reborn/rend3 into closest-filter

Noxime committed:
Change sampler selection into a material flag

Currently the glft loader chooses the sampler filter based on albedo
texture filter, ideally we would have per-texture samplers but that can
be left for later, as well as other sampling methods beyond nearest and
bilinear

cwfitzgerald reviewed (approved):
Thanks so much!
